src/main/java/com/metamx/common/StreamUtils.java

```diff
@@ -44,9 +48,12 @@ public static void copyToFileAndClose(InputStream is, File file) throws IOExcept
     finally {
       CloseQuietly.close(is);
     }
+    return FileUtils.FileCopyResult.buildResult(file.toPath());
```

Comment:
I am concerned about the addition of buildResult here, since this will cause every single copy to read the data twice for the copy and for the checksum, instead of once as before, which could be expensive when dealing with large files and cause performance regressions.